usaspending_api/agency/v2/views/sub_agency.py

```diff
@@ -0,0 +1,138 @@
+from elasticsearch_dsl import A
+from rest_framework.request import Request
+from rest_framework.response import Response
+from typing import Any
+from usaspending_api.agency.v2.views.agency_base import AgencyBase, PaginationMixin
+from fiscalyear import FiscalYear
+from usaspending_api.common.cache_decorator import cache_response
+from usaspending_api.common.elasticsearch.search_wrappers import TransactionSearch
+from usaspending_api.common.helpers.generic_helper import get_pagination_metadata
+from usaspending_api.common.query_with_filters import QueryWithFilters
+
+
+class SubAgencyList(PaginationMixin, AgencyBase):
+    """
+    Obtain the list of subagencies and offices based on the provided toptier_code and fiscal year, as well as award type.
+    Can be either by funding agency or awarding agency.
+    Based on transaction data.
+    """
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/agency/toptier_code/sub_agency.md"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.params_to_validate = ["fiscal_year", "agency_type", "award_type"]
```

Comment:
Currently `award_type_codes` don't appear to be working, but I think after this change it will since TinyShield creates a new copy of the query_params.

Reply:
should be fixed